python/ray/train/_internal/checkpoint.py

```diff
@@ -102,14 +102,26 @@ def _process_checkpoint(
         """Ray Train entrypoint. Perform all processing for a checkpoint."""
         # Get checkpoint from first worker.
         checkpoint_data = checkpoint_results[0].data
-
-        # Decode checkpoint.
-        checkpoint_data = decode_checkpoint_fn(checkpoint_data)
+        checkpoint_metadata = checkpoint_results[0].metadata or {}
+        encoded = checkpoint_results[0].encoded
+
+        # Decode checkpoint if it's not persisted
+        if encoded:
```

Comment:
can we just remove encoding/decoding logic from here and implement it only for `TorchCheckpoint`?

Reply:
Like if we move encoding to `TorchCheckpoint.from_model` and the decoding to `TorchCheckpoint.get_model` then I think we can remove a lot of the encoding logic from our internal plumbing. Just keep it isolated to the Checkpoint classes.

Reply:
@amogkam I have made this a special case in the TorchCheckpoint. Another option would be to make this a developer API in the base Checkpoint class, but I am not sure if there will be many cases where we'll require this sort of behavior.